legend-depot-store-notifications/src/test/java/org/finos/legend/depot/store/notifications/store/mongo/TestQueueMongo.java

```diff
@@ -217,14 +218,29 @@ public void canGetFirstInQueueByFIFO()
         queue.push(event);
         queue.push(event1);
 
-        Optional<MetadataNotification> first = queue.getFirstInQueue();
-        Optional<MetadataNotification> second = queue.getFirstInQueue();
+        Optional<MetadataNotification> first = queue.getFirstInQueueByPriority();
+        Optional<MetadataNotification> second = queue.getFirstInQueueByPriority();
 
         Assert.assertEquals(VERSION,first.get().getVersionId());
         Assert.assertEquals("1.0.1",second.get().getVersionId());
         Assert.assertTrue(first.get().getCreated().before(second.get().getCreated()));
     }
 
+    @Test
+    public void canGetFirstInQueueByPriority()
+    {
+        MetadataNotification event = new MetadataNotification(TESTPROJECT, TEST, TEST, VERSION, null, null, null, null, null, null, null, null, null, EventPriority.LOW);
+        MetadataNotification event1 = new MetadataNotification(TESTPROJECT, TEST, TEST,"1.0.1", null, null, null, null, null, null, null, null, null, EventPriority.HIGH);
+        queue.push(event);
+        queue.push(event1);
+
+        Optional<MetadataNotification> first = queue.getFirstInQueueByPriority();
+        Optional<MetadataNotification> second = queue.getFirstInQueueByPriority();
+
+        Assert.assertEquals("1.0.1", first.get().getVersionId());
+        Assert.assertEquals(VERSION, second.get().getVersionId());
+    }
```

Comment:
you ned to have more scenarios, like two events with same priority and differnt creation times